pennylane_lightning/core/src/simulators/lightning_tensor/tncuda/TNCuda.hpp

```diff
@@ -252,6 +252,31 @@ class TNCuda : public TNCudaBase<PrecisionT, Derived> {
                                                         host_data_size);
     }
 
+    /**
+     * @brief Check if the provided MPS has the correct dimension for C++
+     * backend.
+     *
+     * @param MPS_shape_source Dimension list of incoming MPS.
+     */
+
+    void MPSShapeCheck(
```

Comment:
Let's move this method to `util`.

Reply:
I checked this suggestion but It will be difficult because `MPSShapeCheck` need access to the private variable `sitesExtents_`. If we want to move  `MPSShapeCheck` to `utils` then we need to create a getter function and duplicate the `sitesExtents_` vector.  What do you think?

Reply:
I agree it would be good to move this to util, and having a getter function to return the `sitesExtens_` vector reference makes sense, right?

Reply:
Done!